contracts/strategies/1inch/base/OneInchStrategy_DAI_X.sol

```diff
@@ -0,0 +1,272 @@
+pragma solidity 0.5.16;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "../interface/IMooniswap.sol";
+import "../interface/IFarmingRewardsV2.sol";
+
+import "../../../base/interface/uniswap/IUniswapV2Router02.sol";
+import "../../../base/interface/IStrategy.sol";
+import "../../../base/interface/IVault.sol";
+import "../../../base/interface/weth/Weth9.sol";
+
+import "../../../base/StrategyBase.sol";
+
+/**
+* This strategy is for DAI / X 1inch LP tokens
+* DAI must be token0, and the other token is denoted X
+*/
+contract OneInchStrategy_DAI_X is IStrategy, BaseUpgradeableStrategyUL {
+
+  using SafeERC20 for IERC20;
+  using SafeMath for uint256;
+
+  address public dai = address(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+  address public weth = address(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+
+  uint256 maxUint = uint256(~0);
+
+  // depositToken0 is DAI
+  address public depositToken1;
+
+  address[] public rewardTokens;
+
+  uint256 slippageNumerator = 9;
+  uint256 slippageDenominator = 10;
+
+  constructor() public BaseUpgradeableStrategyUL() {}
+
+  function initializeBaseStrategy(
+    address _storage,
+    address _underlying,
+    address _vault,
+    address _rewardPool,
+    uint256 _profitSharingNumerator
+  ) public initializer {
+
+    BaseUpgradeableStrategyUL.initialize(
+      _storage,
+      _underlying,
+      _vault,
+      _rewardPool,
+      weth,
+      _profitSharingNumerator,  // profit sharing numerator
+      1000, // profit sharing denominator
+      true, // sell
+      0, // sell floor
+      12 hours, // implementation change delay
+      address(0x7882172921E99d590E097cD600554339fBDBc480) // UL Registry
+    );
+
+    require(IVault(_vault).underlying() == _underlying, "vault does not support the required LP token");
+    depositToken1 = IMooniswap(_underlying).token1();
```

Comment:
As this strategy is upgradeable, we will need to save this variable to a dedicated storage slot, like we do in other upgradeable strategies. So that means defining a storage slot, and making `_setDepositToken1()` and `depositToken1()` functions. Examples for this can be found at the start and end of any upgradeable base strategy.